RenderManager_xbox/DistortionCorrectRenderManager.cpp

```diff
@@ -0,0 +1,760 @@
+/** @file
+    @brief Example program that uses the OSVR direct-to-display interface
+           and OpenGL to render a scene with low latency.
+
+    @date 2015
+
+    @author
+    Russ Taylor working through ReliaSolve.com for Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/ClientKit/Context.h>
+#include <osvr/ClientKit/InterfaceStateC.h>
+#include <osvr/Client/RenderManagerConfig.h>
+#include "osvr/RenderKit/RenderManager.h"
+#include "font.h" // Simple helper functions to generate and draw OpenGL bitmapped text
+
+// Needed for render buffer calls.  OSVR will have called glewInit() for us
+// when we open the display.
+#include <GL/glew.h>
+
+// Library/third-party includes
+#ifdef _WIN32
+    #define WIN32_LEAN_AND_MEAN
+    #include <windows.h>
+#endif
+#include <GL/GL.h>
+#include <GL/GLu.h>
+
+// Standard includes
+#include <iostream>
+#include <string>
+#include <stdlib.h> // For exit()
+#include <chrono>
+#include <thread>
+
+//This must come after we include <GL/GL.h> so its pointer types are defined.
+#include "osvr/RenderKit/GraphicsLibraryOpenGL.h"
+
+// Forward declarations of rendering functions defined below.
+void draw_cube(double radius);
+
+// Set to true when it is time for the application to quit.
+// Handlers below that set it to true when the user causes
+// any of a variety of events so that we shut down the system
+// cleanly.  This only works on Windows, but so does D3D...
+static bool quit = false;
+
+// Get an OSVR client context to use to access the devices
+// that we need.
+static osvr::clientkit::ClientContext context(
+  "com.osvr.renderManager.distortionCorrectRenderManager");
+
+static osvr::renderkit::RenderManager *render = nullptr;
+
+// Quadric and sphere to use for rendering.
+static GLUquadric *sphere = nullptr;
+
+// X,Y location
+typedef struct {
+  double x;
+  double y;
+} XY;
+
+static std::vector<float> params;  //< Distortion parameters
+static int activeParam = 0;  //< Which parameter are we adjusting?
+
+static std::string osvrGetString(OSVR_ClientContext context, const std::string& path)
+{
+  size_t len;
+  if (osvrClientGetStringParameterLength(context, path.c_str(), &len) == OSVR_RETURN_FAILURE) {
+    std::string msg = std::string("Couldn't get osvr string length for path ") + path;

```

Comment:
Nothing wrong with this, but you can avoid creating msg var and just use `std::cerr << "Message " << path << std::endl`.